ui/src/pages/CreateAccount.js

```diff
@@ -1,65 +1,307 @@
     return password.length >= MIN_PASSWORD_LENGTH;
   };
 
+  // This function will be called after the user confirms their secret recovery phrase.
   const onClickCreateAccount = async () => {
-    console.log("[INFO]: Creating account.")
-    if (isValid(password)) {
-      await sendPassword(password);
+    console.log("[INFO]: Creating account.");
+    await props.sendPassword(password);
+    setPassword('');
+    setCurrentTab(FINAL_TAB);
+  };
+
+  // This function navigates back depending on the current page.
+  const goBack = async () => {
+    if (currentTab === PASSWORD_TAB) {
+      console.log("[INFO]: Going back to Create or Recovery Page.")
+      await props.restartServer();
+    } else if (currentTab === SHOW_PHRASE_TAB) {
+      console.log("[INFO]: Going back to Password Page.")
       setPassword('');
-      setCreatedAccount(true);
+      setConfirmPassword('');
+      setIsValidPassword(false);
+      setRecoveryPhraseConfirmed(false);
+      setCurrentTab(PASSWORD_TAB);
+    } else if (currentTab === CONFIRM_PHRASE_TAB) {
+      console.log("[INFO]: Going back to View Recovery Phrase Page.");
+      setSelectedRecoveryPhrase([]);
+      appWindow.setSize(DEFAULT_WINDOW_SIZE);
+      setCurrentTab(SHOW_PHRASE_TAB);
     }
-  };
+  }
+
+  const goForward = async () => {
+
+    if (currentTab === PASSWORD_TAB) {
+      setCurrentTab(SHOW_PHRASE_TAB);
+    } else if (currentTab === SHOW_PHRASE_TAB) {
+      appWindow.setSize(CONFIRM_PHRASE_WINDOW_SIZE);
+      setCurrentTab(CONFIRM_PHRASE_TAB);
+    } else if (currentTab === CONFIRM_PHRASE_TAB) {
+      // Recovery Phrase has already been confirmed here, the button will stop being disabled
+      // Once the user has entered it in the correct order.
+      onClickCreateAccount();
+      appWindow.setSize(DEFAULT_WINDOW_SIZE);
+      setCurrentTab(LOADING_TAB);
+    }
+  }
 
-  const onClickConfirmRecoveryPhrase = async () => {
-    console.log("[INFO]: Confirming recovery phrase.")
-    await endInitialConnectionPhase();
+  // This function is called when the user clicks "Finish" at the final page
+  // This function will redirect the user to log in with the password they just made.
+  const onClickFinishSetup = async () => {
+    console.log("[INFO]: Finishing Setup.")
+    await props.restartServer(true);
   };
 
+  // This function enables the Next button to continue in the account creation
+  // process once the user has read the recovery phrase.
+  const onClickConfirmRecoveryPhrase = () => {
+    setRecoveryPhraseConfirmed(true);
+  }
+
+  useEffect(() => {
+
+    if ((password === confirmPassword) && !passwordsMatch) {
+      setPasswordsMatch(true);
+    } else if (!(password === confirmPassword) && (passwordsMatch)) {
+      setPasswordsMatch(false);
+    }
+
+    if (isValid(password) && !isValidPassword) {
+      setIsValidPassword(true);
+    } else if (!isValid(password) && isValidPassword) {
+      setPasswordsMatch(false);
+    }
```

Comment:
not sure why you need the conditions on the right side of these 4 `&&`'s

Reply:
Also, just for readability purposes, it's good to turn code blocks in `useEffect` like this into functions